### PR TITLE
add plugin management utils to node Automation API

### PR DESCRIPTION
**Pull request by EvanBoyle**
Noticed that I missed this during the initial impl. Also fixed an issue that existed in Go with `plugin rm` that I discovered during testing (a missing `--yes` flag).

### CODE REVIEW

**Review by justinvp — APPROVED** [review]
LGTM otherwise.

Nit: I noticed all the commits were merged into master from your last PR. Would you mind "squashing and merging" when merging PRs going forward to keep the history clean (so master doesn't include all the WIP commits)?

File: sdk/nodejs/x/automation/localWorkspace.ts [hunk 1]
```diff
@@ -453,6 +453,39 @@ export class LocalWorkspace implements Workspace {
...
         return stacks;
     }
+    /**
+     * Installs a plugin in the Workspace, for example to use cloud providers like AWS or GCP.
+     *
+     * @param name the name of the plugin.
+     * @param version the version of the plugin e.g. "v1.0.0".
+     */
+    async installPlugin(name: string, version: string): Promise<void> {
+        await this.runPulumiCmd(["plugin", "install", "resource", name, version]);
```
[the remark this answers is not in the document]
Comment (EvanBoyle) [reply sdk/nodejs/x/automation/localWorkspace.ts:463]: I consider this to be another instance of CLI UX not needing to map 1:1 to Automation API UX. I will alter this slightly such that kind has a default value and that the RM params are optional. 

As for the merge, I did this intentionally. I try to keep the commits history fairly clean as I work and find that for bigger changes like this the history is helpful in debugging issues or reconciling decisions later on. Happy to squash in the future if this doesn't align with your sensibilities.

File: sdk/nodejs/x/automation/workspace.ts [hunk 2]
```diff
@@ -171,16 +171,18 @@ export interface Workspace {
      *
      * @param name the name of the plugin.
      * @param version the version of the plugin e.g. "v1.0.0".
+     * @param kind the kind of plugin e.g. "resource"
      */
-    installPlugin(name: string, version: string): Promise<void>;
+    installPlugin(name: string, version: string, kind?: string): Promise<void>;
```
Comment (justinvp) [inline sdk/nodejs/x/automation/workspace.ts:176]: Why not strongly type as `PluginKind`?

Comment (EvanBoyle) [reply sdk/nodejs/x/automation/workspace.ts:176]: Ah sorry, looks like I missed this! I will follow up on this in my next changeset.

**Pull request merged**